src/conv/invokers/mlir_impl_gemm.cpp

```diff
@@ -104,24 +104,24 @@ InvokerFactory MakeMlirFwdInvokerFactory(const ConvolutionContext& ctx)
                             out_strides);
 
     MlirConvArgs args =
-        MakeMlirConvArgs(in_dims, in_strides, weights_dims, weights_strides, out_dims, out_strides);
+        makeMlirConvArgs(in_dims, in_strides, weights_dims, weights_strides, out_dims, out_strides);
 
     return [=](const std::vector<Kernel>& kernels) {
         return [=](const Handle& handle, const AnyInvokeParams& primitive_parameters) mutable {
             const auto& data_ctx = primitive_parameters.CastTo<conv::DataInvokeParams>();
             const auto& tensors  = data_ctx.tensors;
 
             // Filter
-            args.filter.basePtr =
-                const_cast<void*>(tensors.w); // NOLINT (cppcoreguidelines-pro-type-const-cast)
-            args.filter.data =
-                const_cast<void*>(tensors.w); // NOLINT (cppcoreguidelines-pro-type-const-cast)
+            // NOLINTNEXTLINE (cppcoreguidelines-pro-type-const-cast)
+            args.filter.basePtr = const_cast<void*>(reinterpret_cast<const void*>(tensors.w));
```

Comment:
You can't reinterpret_cast a `cl_mem` object to a `void*` to get a pointer. You need to use [`clGetMemObjectInfo`](https://www.khronos.org/registry/OpenCL/sdk/1.0/docs/man/xhtml/clGetMemObjectInfo.html).

Reply:
Thanks for figuring this out. Will do the below to address this. Let me know if I understand this right.

```cpp
#if MIOPEN_BACKEND_OPENCL
  void* p;
  clGetMemObjectInfo(tensors.w, CL_MEM_HOST_PTR, sizeof(p), &p, NULL);
  args.filter.basePtr = p;
#else
  args.filter.basePtr = const_cast<void*>(tensors.w);
#endif
```

Reply:
Yes that looks correct.